tests/jest/unit/assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.test.js

```diff
@@ -0,0 +1,228 @@
+					DirectionMode,
+				} );
+			}
+		} );
+
+		$e.components.register( new class extends $e.modules.ComponentBase {
+			getNamespace() {
+				return 'panel';
+			}
+
+			defaultRoutes() {
+				return {
+					'editor/advanced': () => {},
+				};
+			}
+		} );
+	} );
+
+	afterEach( async () => {
+		await freeMock();
+	} );
+
+	it( 'Should not set direction for containers that don\'t support UI states', async () => {
+		// Arrange.
+		const container = createContainer();
+
+		// Act.
+		$e.hooks.runUIAfter( 'document/elements/settings', { container } );
+
+		// Assert.
+		expect( $e.uiStates.getCurrent( 'document/direction-mode' ) ).toBeFalsy();
+	} );
+
+	it( 'Should set direction for containers that support UI states', async () => {
+		// Arrange.
+		const container = createContainer( {
+			renderer: {
+				view: {
+					getCurrentUiStates: () => ( {
+						directionMode: 'column',
+					} ),
+				},
+			},
+		} );
+
+		// Act.
+		$e.hooks.runUIAfter( 'document/elements/settings', { container } );
+
+		// Assert.
+		expect( $e.uiStates.getCurrent( 'document/direction-mode' ) ).toBe( 'column' );
+	} );
+
+	it( 'Should use parent direction when editing the advanced settings', async () => {
+		// Arrange.
+		const child = createContainer( {
+			renderer: {
+				view: {
+					getCurrentUiStates: () => {},
+				},
+			},
+		} );
+
+		const parent = createContainer( {
+			renderer: {
+				view: {
+					getCurrentUiStates: () => ( {
+						directionMode: 'row',
+					} ),
+				},
+			},
+			children: [ child ],
+		} );
+
+		$e.route( 'panel/editor/advanced' );
+
+		// Act.
+		$e.hooks.runUIAfter( 'document/elements/settings', { container: child } );
+
+		// Assert.
+		expect( $e.uiStates.getCurrent( 'document/direction-mode' ) ).toBe( 'row' );
+	} );
+
+	it( 'Should remove the direction mode UI state when the container doesn\'t support it', async () => {
+		// Arrange.
+		const container = createContainer( {
+			renderer: {
+				view: {
+					getCurrentUiStates: () => ( {} ),
+				},
+			},
+		} );
+
+		$e.uiStates.set( 'document/direction-mode', 'row' );
+
+		// Act.
+		$e.hooks.runUIAfter( 'document/elements/settings', { container } );
+
+		// Assert.
+		expect( $e.uiStates.getCurrent( 'document/direction-mode' ) ).toBeFalsy();
+	} );
+
+	it( 'Should work with multiple containers', async () => {
+		// Arrange.
+		const container1 = createContainer( {
+			renderer: {
+				view: {
+					getCurrentUiStates: () => ( {
+						directionMode: 'column',
+					} ),
+				},
+			},
+		} );
+
+		const container2 = createContainer( {
+			renderer: {
+				view: {
+					getCurrentUiStates: () => ( {
+						directionMode: 'row',
+					} ),
+				},
+			},
+		} );
+
+		// Act.
+		$e.hooks.runUIAfter( 'document/elements/settings', { containers: [ container1, container2 ] } );
+
+		// Assert.
+		expect( $e.uiStates.getCurrent( 'document/direction-mode' ) ).toBe( 'row' );
+	} );
+} );
+
+/**
+ * Mock Container.
+ * TODO: Move to testing utils.
```

Comment:
Hm just suggestion,  do you think the way I did is good? If so, let use it.

`elementor/tests/jest/unit/modules/web-cli/assets/js/core/mock/api` utils for create API mock.
`modules/web-cli/assets/js/api.js` original path